crates/ark-contracts/arkchain/src/order/order_v1.cairo

```diff
@@ -95,10 +95,15 @@ impl OrderTraitOrderV1 of OrderTrait<OrderV1> {
             return Result::Err(OrderValidationError::AdditionalDataTooLong);
         }
 
+        if (*self.broker_id != starknet::get_caller_address().into()) {
+            return Result::Err(OrderValidationError::MismatchBroker);
+        }
```

Comment:
it shouldn't be the caller address here, but check if the broker_id contain in the current order is in the storage